### PR TITLE
Scripts/HallsOfLightning: Boss Volkhan - Add OOC Forgecast

### LINKED ISSUE #14066: [Instance/HoL] Volkhan - Missing Visual Spell
http://de.wowhead.com/npc=28587/volkhan

This guy should cast every 15 sec. this spell:
ID - 52654

on his forge.

DoCastAOE() and DoCast() does not seem to work for this spell.

Currently this Visual is missing.

### PR DESCRIPTION
**Pull request by Lumber-Jack-2**
closes #14066

### CODE REVIEW

File: src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp [hunk 1]
```diff
@@ -29,7 +29,7 @@ EndScriptData */
 #include "Player.h"
 #include "SpellInfo.h"
 
-enum Enums
+enum VolkhanTexts
```
Comment (Gooyeth) [inline src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp]: Texts

File: src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp [hunk 2]
```diff
@@ -38,29 +38,50 @@ enum Enums
     SAY_DEATH                               = 4,
     EMOTE_TO_ANVIL                          = 5,
     EMOTE_SHATTER                           = 6,
+};
 
+enum VolkhanSpells
```
Comment (Gooyeth) [inline src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp]: enum Spells

File: src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp [hunk 3]
```diff
@@ -38,29 +38,50 @@ enum Enums
...
     SPELL_SUMMON_MOLTEN_GOLEM               = 52405,
+    SPELL_FORGE_VISUAL                      = 52654,
 
     // Molten Golem
     SPELL_BLAST_WAVE                        = 23113,
     SPELL_IMMOLATION_STRIKE                 = 52433,
     SPELL_SHATTER                           = 52429,
+};
+
+enum VolkhanEvents
```
Comment (Gooyeth) [inline src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp]: enum Events

File: src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp [hunk 4]
```diff
@@ -38,29 +38,50 @@ enum Enums
...
+    EVENT_SHATTERING_STOMP                  = 2,
+    EVENT_SHATTER                           = 3,
+    EVENT_FORGE_CAST                        = 4,
 
+    // Molten Golem
+    EVENT_BLAST                             = 5,
+    EVENT_IMMOLATION                        = 6
+};
+
+enum VolkhanCreatures
```
Comment (Gooyeth) [inline src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp]: enum Npcs

File: src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp [hunk 5]
```diff
@@ -38,29 +38,50 @@ enum Enums
...
     NPC_VOLKHAN_ANVIL                       = 28823,
     NPC_MOLTEN_GOLEM                        = 28695,
     NPC_BRITTLE_GOLEM                       = 28681,
-
     MAX_GOLEM                               = 2,
-
     DATA_SHATTER_RESISTANT                  = 2042
 };
 
+enum VolkhanPhases
```
Comment (Gooyeth) [inline src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp]: enum Phases

File: src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp (hunk 1 above)
Comment (DDuarte) [inline src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp]: Doesn't really matter, does it?

Comment (Gooyeth) [inline src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp]: Codestyle, the name is already in the script.

File: src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp [hunk 6]
```diff
@@ -83,37 +104,16 @@ class boss_volkhan : public CreatureScript
...
-
-        uint8 GolemsShattered;
-        uint32 m_uiPause_Timer;
-        uint32 m_uiShatteringStomp_Timer;
-        uint32 m_uiShatter_Timer;
-        uint32 m_uiDelay_Timer;
-        uint32 m_uiSummonPhase;
-
-        uint32 m_uiHealthAmountModifier;
+        }
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp]: white spaces

File: src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp [hunk 7]
```diff
@@ -83,37 +104,16 @@ class boss_volkhan : public CreatureScript
         }
 
         void Initialize()
-        {
-            m_bIsStriking = false;
-            m_bHasTemper = false;
+        {
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp]: white space here too

File: src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp [hunk 8]
```diff
@@ -121,14 +121,18 @@ class boss_volkhan : public CreatureScript
...
         }
 
         void EnterCombat(Unit* /*who*/) override
         {
             Talk(SAY_AGGRO);
-
+            events.SetPhase(PHASE_NORMAL);
+            events.ScheduleEvent(EVENT_PAUSE,            3.5 * IN_MILLISECONDS, 0, PHASE_NORMAL);
+            events.ScheduleEvent(EVENT_SHATTERING_STOMP,   0 * IN_MILLISECONDS, 0, PHASE_NORMAL);
+            events.ScheduleEvent(EVENT_SHATTER,            5 * IN_MILLISECONDS, 0, PHASE_NORMAL);
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/Ulduar/HallsOfLightning/boss_volkhan.cpp]: these spaces look ugly imo, but just cosmetic :P

**Comment by MitchesD** [comment]
use BossAI, I have no idea why there is still ScriptedAI, with boss AI you use _Reset() _EnterCombat() methods etc.
also there is missing events.Reset() so another reason why you should use BossAI

**Comment by Rushor** [comment]
okay done :)

**Comment by jackpoz** [comment]
This broke EVENT_SHATTERING_STOMP event since it's executed only once at start, the health check fails and it's never rescheduled

**Pull request merged**